flatland_server/src/simulation_manager.cpp

```diff
@@ -113,6 +104,19 @@ void SimulationManager::Main() {
   ROS_INFO_NAMED("SimMan", "Simulation loop started");
 
   while (ros::ok() && run_simulator_) {
+    if (service_manager_ == nullptr) {
+      try {
+        world_->LoadWorldEntities(world_yaml_file_);
+        if (show_viz_) {
+          world_->DebugVisualize();
+        }
+        service_manager_ =
+            std::unique_ptr<ServiceManager>(new ServiceManager(this, world_));
+      } catch (const YAMLException& ex) {
+        ROS_DEBUG_STREAM("Tried to load world yaml file " << world_yaml_file_);
+      }
+    }
+
```

Comment:
Can you do things in the loop if the service_manager is not set (i.e. an exception occurred)? If not, this code may be moved before the main loop.